File: src/contests/cg/puzzle/csharp/the-fastest.cs
// 🎮 CodinGame Puzzle - the-fastest
// https://www.codingame.com/training/medium/the-fastest

using System;

class Solution
{
    static void Main()
    {
        // Times are HH:MM:SS so lexicographic comparison equals chronological comparison.
        int n = int.Parse(Console.ReadLine().Trim());
        string best = "";
        for (int i = 0; i < n; i++)
        {
            string t = Console.ReadLine().Trim();
            if (best == "" || string.CompareOrdinal(t, best) < 0) best = t;
        }
        Console.WriteLine(best);
    }
}

File: src/contests/cg/puzzle/fsharp/the-fastest.fs
// 🎮 CodinGame Puzzle - the-fastest
// https://www.codingame.com/training/medium/the-fastest

// Times are HH:MM:SS so lexicographic comparison equals chronological comparison.
let n = int (System.Console.ReadLine().Trim())
let mutable best = ""
for _ in 1 .. n do
    let t = System.Console.ReadLine().Trim()
    if best = "" || System.String.CompareOrdinal(t, best) < 0 then best <- t
printfn "%s" best
